content/blog/rn-bridging-native-modules/index.md

```diff
@@ -0,0 +1,341 @@
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+```
+
+This moves the code that loads the URL from `init()` to the `setWebViewURL()` method which is automatically called when the `webViewURL` prop is set. This is because at the time when `init()` is run the value from the prop hasn't been set yet.
+
+Try running the updated code and it should still work while passing the URL as a prop from React Native!
+
+## Passing data back from Swift class to React Native
+
+The final step of this guide will be to pass data from Swift back to React Native. For this example whenever the WebView redirects, it will pass the URL to React Native. To do this we need to make use of events.  
+
+In our Swift class again: 
+
+```swift
+// MyWebView.swift
+
+...
+
+@objc(MyWebView)
+public class MyWebView: UIView, WKNavigationDelegate, WKUIDelegate {
+    ...
+
+    @objc var onRedirect: RCTDirectEventBlock? // Add
+
+    ...
+
+    // Add
+    public func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: (WKNavigationActionPolicy) -> Void) {
+      onRedirect!(["url": navigationAction.request.url?.absoluteString ?? ""])
+      decisionHandler(.allow)
+    }
+}
+```
+
+- We add another property to the `MyWebView` class, this time `onRedirect()`, which is of type `RCTDirectEventBlock`.  
+- Then we implement the `webView` method that decides the policy for navigation actions, e.g. redirects. In this case, we simply allow every redirect to trigger the `onRedirect()` event to send the URL back to React Native.  
+
+We need to update our Objective-C **MyWebView.m** file to expose the new property, same as before:
+
+```m
+// MyWebView.m
+#import <Foundation/Foundation.h>
+#import <React/RCTBridgeModule.h>
+#import <React/RCTViewManager.h>
+
+@interface RCT_EXTERN_MODULE(MyWebViewManager, RCTViewManager)
+
+RCT_EXPORT_VIEW_PROPERTY(webViewURL, NSString)
+RCT_EXPORT_VIEW_PROPERTY(onRedirect, RCTDirectEventBlock)
+
+@end
+```
+
+We also need to add another couple of imports to our bridging header file, to allow the usage of events:  
+
+```h
+// [ProjectName]-Bridging-Header.h
+#import <React/RCTBridgeModule.h>
+#import <React/RCTViewManager.h>
+#import <React/RCTEventDispatcher.h> // add this
+#import "React/RCTEventEmitter.h"    // add this
+```
+
+Finally, back to our React Native where we will intercept the event, and log the URL to the console:  
+
+```js
+    <WebView
+        nativeConfig={{
+            component: MyWebView,
+            props: {
+                webViewURL: "https://beanloop.se",
+                onRedirect: ({nativeEvent}) => {
+                    console.log(nativeEvent.url) // [redirect url]
+                },
+            },
+            viewManager: MyWebViewManager,
+        }}
+    />
+```  
+
+We can see the updated WebView in action below:
+
+![WebView in action example 2](./rn-webview-example-2.png)
+
+# Done!
```

Comment:
At the end here, you could say something like, watch out for the next post where I'll be focusing on bridging from Java for android